src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksWebServiceCallsServiceImpl.java

```diff
@@ -397,14 +402,31 @@ private Invocation buildMultiPartRequestForSupplierUpload(Client client, URI uri
         multiPart.setMediaType(MediaType.MULTIPART_FORM_DATA_TYPE);
         multiPart.bodyPart(csvPart);
         
-        return client.target(uri)
-                .request()
+        WebTarget target = client.target(uri);
+        Invocation.Builder requestBuilder = target.request();
+        disableRequestChunkingIfNecessary(client, requestBuilder);
+        
+        return requestBuilder
                 .accept(MediaType.APPLICATION_JSON_TYPE)
                 .header(PaymentWorksWebServiceConstants.AUTHORIZATION_HEADER_KEY, 
                              PaymentWorksWebServiceConstants.AUTHORIZATION_TOKEN_VALUE_STARTER + getPaymentWorksAuthorizationToken())
                 .buildPost(Entity.entity(multiPart, MediaType.MULTIPART_FORM_DATA_TYPE));
     }
 
+    private void disableRequestChunkingIfNecessary(Client client, Invocation.Builder requestBuilder) {
+        if (client instanceof org.apache.cxf.jaxrs.client.spec.ClientImpl) {
```

Comment:
I thought this was a problem only when the request was over a certain size, like 30 vendors or so.  Should we disable chunking only when the instance is as coded above and the size limit is reached?

Reply:
The point is that we never want the data to be sent in a chunked manner, since the PaymentWorks endpoint cannot handle that. This fix just forces the JAX-RS client to always use non-chunked mode, rather than letting CXF decide whether or not to chunk the data.

Also, at this point in the code, the data is in an InputStream so the number of rows cannot be easily derived. If we wanted to add a row-count-related restriction, we would need to update the method signatures to accept new or different arguments accordingly.